uniswap-v2-core/contracts/VexchangeV2Factory.sol

```diff
@@ -24,9 +24,10 @@ contract VexchangeV2Factory is IVexchangeV2Factory, Ownable {
     event DefaultPlatformFeeChanged(uint oldDefaultPlatformFee, uint newDefaultPlatformFee);
     event DefaultRecovererChanged(address oldDefaultRecoverer, address newDefaultRecoverer);
 
-    constructor(uint _defaultSwapFee, uint _defaultPlatformFee, address _defaultRecoverer) public {
+    constructor(uint _defaultSwapFee, uint _defaultPlatformFee, address _platformFeeTo, address _defaultRecoverer) public {
         defaultSwapFee = _defaultSwapFee;
         defaultPlatformFee = _defaultPlatformFee;
+        platformFeeTo = _platformFeeTo;
```

Comment:
Should we add a `require` that we never set `platformFeeTo` to `address(0)`?

Reply:
Should `_platformFeeTo` come after `_defaultRecoverer` to replicate the ordering of member declarations just above:
```solidity
    uint public defaultSwapFee;
    uint public defaultPlatformFee;
    address public defaultRecoverer;

    address public platformFeeTo;
```

Or it might be better to just change the member declaration order to:
```solidity
    uint public defaultSwapFee;
    uint public defaultPlatformFee;
    address public platformFeeTo;
    address public defaultRecoverer;
```

Reply:
> Should we add a `require` that we never set `platformFeeTo` to `address(0)`?

Yes I thought of that too, because I believe in defensive programming and checking and asserting whenever we can. In this case should we check for address(0) for the defaultRecoverer too?

Reply:
> Should `_platformFeeTo` come after `_defaultRecoverer` to replicate the ordering of member declarations just above:
> 
> ```solidity
>     uint public defaultSwapFee;
>     uint public defaultPlatformFee;
>     address public defaultRecoverer;
> 
>     address public platformFeeTo;
> ```
> 
> Or it might be better to just change the member declaration order to:
> 
> ```solidity
>     uint public defaultSwapFee;
>     uint public defaultPlatformFee;
>     address public platformFeeTo;
>     address public defaultRecoverer;
> ```

Yes I think the latter makes more sense

Reply:
> > Should we add a `require` that we never set `platformFeeTo` to `address(0)`?
> 
> Yes I thought of that too, because I believe in defensive programming and checking and asserting whenever we can. In this case should we check for address(0) for the defaultRecoverer too?

And following up from that, should we check for address(0) in `setPlatformFeeTo`, `setRecovererForPair`, and `setDefaultRecoverer`?

Reply:
L260 in VexchangeV2Pair.sol lets us disable the `recoverToken` method which may be convenient / useful. So given its safe to set to zero im happy to not add the asserts there.
```solidity
require(recoverer != address(0), "VexchangeV2: RECOVERER_ZERO_ADDRESS");
```

However, for `setPlatformFeeTo` I do think it would be a good change to check we do not set it to the zero address

Reply:
Agreed. I think this approach is reasonable.